syntax = "proto3";  // 声明了protobuf的版本

package fixbug; // 声明了代码所在的包（对于c++来说是namespace)

option cc_generic_services = true;

message resultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

// 定义登录请求消息类型 name pwd
message LoginRequest
{
    string name = 1;
    string pwd = 2;
}

// 定义登录消息类型
message LoginResponse
{
    resultCode result = 1;  
    bool success = 2;
}

message GetFriendListRequest
{
    uint32 userid = 1;
}

message User
{
    bytes name = 1;
    uint32 age = 2;
    enum Sex
    {
        man = 0;
        woman = 1;
    }
    Sex sex = 3;
}

message GetFriendListResponse
{
    resultCode result = 1;
    repeated User friendList = 2;   // 定义了一个列表类型
}

// 在protobuf里面怎么定义描述rpc方法的类型 - service
service UserServiceRpc
{
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc GetFriendList (GetFriendListRequest) returns (GetFriendListResponse);
}